### PR TITLE
PIM-8013: Fix 401 redirection on non-authorized page

### PR DESCRIPTION
**Pull request by anaelChardan**
**Description (for Contributor and Core Developer)**

The current problem is that we currently have problems like that: 

```
benoit@tec:/tmp$ wget http://pim-ce-master.local/-2019-01-23 18:35:19- http://pim-ce-master.local/Resolving pim-ce-master.local (pim-ce-master.local)... 127.0.0.1Connecting to pim-ce-master.local (pim-ce-master.local)|127.0.0.1|:80... connected.HTTP request sent, awaiting response... 401 Unauthorized
```

That what we get now:

```
wget http://localhost:8081
--2019-05-10 17:07:17--  http://localhost:8081/
Resolving localhost (localhost)... ::1, 127.0.0.1
Connecting to localhost (localhost)|::1|:8081... connected.
HTTP request sent, awaiting response... 302 Found
Location: http://localhost:8081/user/login [following]
--2019-05-10 17:07:24--  http://localhost:8081/user/login
Reusing existing connection to [localhost]:8081.
HTTP request sent, awaiting response... 200 OK
Length: unspecified [text/html]
Saving to: ‘index.html.2’

index.html.2                                    [ <=>                                                                                      ]   3.47K  --.-KB/s    in 0s

2019-05-10 17:07:24 (386 MB/s) - ‘index.html.2’ saved [3558]
```

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | -
| Added legacy Behats               | -
| Added acceptance tests            | -
| Added integration tests           | -
| Changelog updated                 | OK
| Review and 2 GTM                  | -
| Micro Demo to the PO (Story only) | -
| Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

File: src/Akeneo/Platform/Bundle/UIBundle/Http/FormAuthenticationEntryPoint.php [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function start(Request $request, AuthenticationException $authException = null)
+    {
+        if ($this->useForward) {
+            //This is the added code
+            if ($request->getRequestUri() === "/") {
```
Comment (BitOne) [inline src/Akeneo/Platform/Bundle/UIBundle/Http/FormAuthenticationEntryPoint.php]: Are you sure you don't have this problem on other URLs (like /user/ogin ?)

Comment (anaelChardan) [reply src/Akeneo/Platform/Bundle/UIBundle/Http/FormAuthenticationEntryPoint.php]: I'll have to test ;) but I don't think so. It's only due to redirect from pages which are not loaded by Symfony directly, that's what I understood at least by discussing with @juliensnz  ;)

Comment (anaelChardan) [reply src/Akeneo/Platform/Bundle/UIBundle/Http/FormAuthenticationEntryPoint.php]: it works : 
```
wget http://localhost:8081/user/login
--2019-05-13 14:52:39--  http://localhost:8081/user/login
Resolving localhost (localhost)... ::1, 127.0.0.1
Connecting to localhost (localhost)|::1|:8081... connected.
HTTP request sent, awaiting response... 200 OK
Length: unspecified [text/html]
Saving to: ‘login’

login                     [ <=>                    ]   3.47K  --.-KB/s    in 0s

2019-05-13 14:52:40 (8.17 MB/s) - ‘login’ saved [3558]
```

Comment (BitOne) [reply src/Akeneo/Platform/Bundle/UIBundle/Http/FormAuthenticationEntryPoint.php]: :+1:

**Pull request merged**